### PR TITLE
Don't emit clippy::useless_conversion on type aliases

### LINKED ISSUE #10773: Don't emit `clippy::useless_conversion` on type aliases
### Description

Currently, `clippy::useless_conversion` will still be emitted even on type aliases, for example:
```rust
HWND::from(0isize)
```
Clippy will suggest:
```rust
0isize
```
This conveys the type worse, but looking over rustc_hir, it doesn't seem whether it's a type alias is available so I don't know if this is possible to implement. I saw a few things relating to whether it's a trait alias but not whether it's a type alias. If this is possible, I'm happy to implement this :slightly_smiling_face:

### Version

```text
rustc 1.71.0-nightly (9ecda8de8 2023-04-30)
binary: rustc
commit-hash: 9ecda8de85ce893cc3fc748ab06be0b8250147a7
commit-date: 2023-04-30
host: x86_64-unknown-linux-gnu
release: 1.71.0-nightly
LLVM version: 16.0.2
```

### Additional Labels

@rustbot label +C-enhancement

### PR DESCRIPTION
**Pull request by Centri3**
Fixes #10773

changelog: Enhancement: [`useless_conversion`]: Don't lint on type aliases

### CODE REVIEW

File: clippy_lints/src/useless_conversion.rs [hunk 1]
```diff
@@ -195,3 +197,12 @@ impl<'tcx> LateLintPass<'tcx> for UselessConversion {
         }
     }
 }
+
+/// `cx.qpath_res` seems to return `AssocFn` so we do this instead
+fn is_ty_alias(qpath: &QPath<'_>) -> bool {
```
Comment (llogiq) [inline clippy_lints/src/useless_conversion.rs:202]: This could well be in `clippy_utils`, I think.

**Review by llogiq — APPROVED** [review]
Thank you! This looks very good. In fact, the only improvement I can suggest is to move the `is_ty_alias` function into `clippy_utils` so we can use it from other lints.

Are you OK with doing that? Otherwise I'll r+ and make a note to do it next time I refactor stuff.

File: clippy_lints/src/useless_conversion.rs (hunk 1 above)
Comment (Centri3) [reply clippy_lints/src/useless_conversion.rs]: done, though I did have to enable the feature `if_let_guards`.

**Comment by llogiq** [comment]
Thank you!

@bors r+

**Pull request merged**